modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

```diff
@@ -626,13 +626,17 @@ private void onDiscoveryEvent(DiscoveryEvent evt, DiscoCache cache) {
                     else {
                         DiscoveryDataClusterState state = cctx.kernalContext().state().clusterState();
 
+                        assert state.transition() : state + " evt: " + evt + " msg: " + customMsg;
```

Comment:
Why do we need this assertion? What does it protect us from?

Reply:
This line was removed in https://github.com/gridgain/gridgain/commit/c932f681e53c1b5d5fe5c637d5e387ded8c6443c#diff-896382ae47330cb2c3aa1505290c99e18c93c338b9e584106d2e9e14e00a4b92 and a ticket GG-30804 was created. So now I revert it.